pkg/client/fake/client_test.go

```diff
@@ -2068,6 +2070,70 @@ var _ = Describe("Fake client", func() {
 		err := cl.Get(context.Background(), client.ObjectKey{Name: "foo"}, obj)
 		Expect(apierrors.IsNotFound(err)).To(BeTrue())
 	})
+
+	It("disallows scale subresources on unsupported built-in types", func() {
+		scheme := runtime.NewScheme()
+		Expect(corev1.AddToScheme(scheme)).To(Succeed())
+		Expect(apiextensions.AddToScheme(scheme)).To(Succeed())
+
+		obj := &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "foo",
+			},
+		}
+		cl := NewClientBuilder().WithScheme(scheme).WithObjects(obj).Build()
+
+		scale := &autoscalingv1.Scale{Spec: autoscalingv1.ScaleSpec{Replicas: 2}}
+		Expect(cl.SubResource(subResourceScale).Get(context.Background(), obj, scale)).ToNot(Succeed())
+		Expect(cl.SubResource(subResourceScale).Update(context.Background(), obj, client.WithSubResourceBody(scale))).ToNot(Succeed())
+	})
+
+	It("should be able to Get scale subresources", func() {
+		obj := &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "deploy",
+			},
+			Spec: appsv1.DeploymentSpec{
+				Replicas: ptr.To[int32](2),
+			},
+		}
+		cl := NewClientBuilder().WithObjects(obj).Build()
+		objOriginal := obj.DeepCopy()
+
+		scale := &autoscalingv1.Scale{}
+		Expect(cl.SubResource(subResourceScale).Get(context.Background(), obj, scale)).NotTo(HaveOccurred())
+
+		actual := &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: obj.Name}}
+		Expect(cl.Get(context.Background(), client.ObjectKeyFromObject(actual), actual)).To(Succeed())
+
+		objOriginal.APIVersion = scale.APIVersion
+		objOriginal.Kind = scale.Kind
```

Comment:
These lines confuse me. Neither `objOriginal` nor `scale` should have either `APIVersion` or `Kind` set - Why is this needed? Same question for the update test.